snuba/query/parser/validation/functions.py

```diff
@@ -38,39 +40,44 @@ def validate(self, exp: Expression, data_source: DataSource) -> None:
         if not isinstance(exp, FunctionCall):
             return
 
-        entity_validators: Mapping[str, FunctionCallValidator] = {}
-        data_model: Optional[ColumnSet] = None
+        # If the data_source is a JoinClause it will have mutiple
+        # entities so we want to make sure we validate them all.
+        entities: List[Entity] = []
+
         if isinstance(data_source, QueryEntity):
             entity = get_entity(data_source.key)
-            entity_validators = entity.get_function_call_validators()
+            entities.append(entity)
+
+        elif isinstance(data_source, JoinClause):
+            alias_map = data_source.get_alias_node_map()
+            for _, node in alias_map.items():
+                assert isinstance(node.data_source, QueryEntity)  # mypy
+                entities.append(get_entity(node.data_source.key))
+
+        elif isinstance(data_source, ProcessableQuery):
+            entity = get_entity(data_source.get_from_clause().key)
+            entities.append(entity)
 
-            common_function_validators = (
-                entity_validators.keys() & default_validators.keys()
-            )
-            if common_function_validators:
```

Comment:
also I got rid of this check because I don't really get why we need it. we don't have any entity_validators atm and it seems to make sense that if you had an entity specific one, you should use that over the default ?